File: Section 3/BigOExercise1.js
// What is the Big O of the below function? (Hint, you may want to go line by line)

function funChallenge(input) {
    let a = 10; //O(1)
    a = 50 + 3; //O(1)
  
    for (let i = 0; i < input.length; i++) //O(n) já que o número de vezes que o for vai ser repetir depende do input
    {
      anotherFunction(); // O(n)
      let stranger = true; //O(n) It will run how many times the input is provided
      a++;//O(n)
    }
    return a; //O(1)
  }
  
  // If we have a mathy look now
  // We'll have the following operation
  // 3 O(1)  + 4 O(N)
  // 3 + n + n + n + n
  // 3 + 4n
  // Our Big O notation is  O(3 + 4n)

File: Section 3/python/exercises/bigOExercise1.py
def funChallenge(input):
    a = 10 #O(1)
    a = 50 + 3 #O(1)

    for i in range(len(input)): # O(n)
        another_function() #O(n)
        stranger = True #O(n)
        a += 1 #O(n)

    return a

def another_function():
    print()

  # If we have a mathy look now
  # We'll have the following operation
  # 3 O(1)  + 4 O(N)
  # 3 + n + n + n + n
  # 3 + 4n
  # Our Big O notation is  O(3 + 4n)
  # Simplifying it would be O(n)